skynet/modules/ttt/llm_selector.py

```diff
@@ -0,0 +1,107 @@
+from typing import Optional
+
+from langchain_community.chat_models import ChatOCIGenAI
+from langchain_core.language_models.chat_models import BaseChatModel
+
+from langchain_openai import AzureChatOpenAI, ChatOpenAI
+
+from skynet.auth.user_info import CredentialsType, get_credentials
+from skynet.env import (
+    app_uuid,
+    azure_openai_api_version,
+    llama_path,
+    oci_auth_type,
+    oci_compartment_id,
+    oci_config_profile,
+    oci_model_id,
+    oci_service_endpoint,
+    openai_api_base_url,
+    use_oci,
+)
+from skynet.logs import get_logger
+from skynet.modules.ttt.summaries.v1.models import Processors
+
+log = get_logger(__name__)
+
+
+class LLMSelector:
+    @staticmethod
+    def get_job_processor(customer_id: str) -> Processors:
+        options = get_credentials(customer_id)
+        secret = options.get('secret')
+        api_type = options.get('type')
+
+        if secret:
+            if api_type == CredentialsType.OPENAI.value:
+                return Processors.OPENAI
+            elif api_type == CredentialsType.AZURE_OPENAI.value:
+                return Processors.AZURE
+
+        # OCI doesn't have a secret since it's provisioned for the instance as a whole.
+        if api_type == CredentialsType.OCI.value:
+            return Processors.OCI
+
+        return Processors.LOCAL
+
+    @staticmethod
+    def select(customer_id: str, max_completion_tokens: Optional[int] = None) -> BaseChatModel:
+        processor = LLMSelector.get_job_processor(customer_id)
+        options = get_credentials(customer_id)
+
+        if processor == Processors.OPENAI:
+            log.info(f'Forwarding inference to OpenAI for customer {customer_id}')
+
+            return ChatOpenAI(
+                api_key=options.get('secret'),
+                max_completion_tokens=max_completion_tokens,
+                model_name=options.get('metadata').get('model'),
+                temperature=0,
+            )
+        elif processor == Processors.AZURE:
+            log.info(f"Forwarding inference to Azure-OpenAI for customer {customer_id}")
+
+            metadata = options.get('metadata')
+
+            return AzureChatOpenAI(
+                api_key=options.get('secret'),
+                api_version=azure_openai_api_version,
+                azure_endpoint=metadata.get('endpoint'),
+                azure_deployment=metadata.get('deploymentName'),
+                max_completion_tokens=max_completion_tokens,
+                temperature=0,
+            )
+        elif processor == Processors.OCI or use_oci:
```

Comment:
shouldn't this become `and oci_available`? or have that check in the `get_job_processor` logic

Reply:
yeah it makes sense!

Reply:
done!